bundles/sirix-core/src/main/java/org/sirix/access/ResourceConfiguration.java

```diff
@@ -0,0 +1,668 @@
+  public Path getConfigFile() {
+    return resourcePath.resolve(ResourcePaths.CONFIG_BINARY.getPath());
+  }
+
+  /**
+   * JSON names.
+   */
+  private static final String[] JSONNAMES =
+      {"revisioning", "revisioningClass", "numbersOfRevisiontoRestore", "byteHandlerClasses", "storageKind", "hashKind",
+          "hashFunction", "compression", "pathSummary", "resourceID", "deweyIDsStored", "persistenter"};
+
+  /**
+   * Serialize the configuration.
+   *
+   * @param config configuration to serialize
+   * @throws SirixIOException if an I/O error occurs
+   */
+  public static void serialize(final ResourceConfiguration config) throws SirixIOException {
+    final Path configFile = config.getConfigFile();
+    try (final FileWriter fileWriter = new FileWriter(configFile.toFile());
+        final JsonWriter jsonWriter = new JsonWriter(fileWriter)) {
+      jsonWriter.beginObject();
+      // Versioning.
+      jsonWriter.name(JSONNAMES[0]);
+      jsonWriter.beginObject();
+      jsonWriter.name(JSONNAMES[1]).value(config.revisioningType.name());
+      jsonWriter.name(JSONNAMES[2]).value(config.numberOfRevisionsToRestore);
+      jsonWriter.endObject();
+      // ByteHandlers.
+      jsonWriter.name(JSONNAMES[3]);
+      jsonWriter.beginArray();
+      for (final ByteHandler handler : config.byteHandlePipeline.getComponents()) {
+        ByteHandlerKind.getKind(handler.getClass()).serialize(handler, jsonWriter);
+      }
+      jsonWriter.endArray();
+      // Storage type.
+      jsonWriter.name(JSONNAMES[4]).value(config.storageType.name());
+      // Hashing type.
+      jsonWriter.name(JSONNAMES[5]).value(config.hashType.name());
+      // Hash function.
+      jsonWriter.name(JSONNAMES[6]).value(config.nodeHashFunction.toString());
+      // Text compression.
+      jsonWriter.name(JSONNAMES[7]).value(config.useTextCompression);
+      // Path summary.
+      jsonWriter.name(JSONNAMES[8]).value(config.withPathSummary);
+      // ID.
+      jsonWriter.name(JSONNAMES[9]).value(config.id);
+      // Dewey IDs stored or not.
+      jsonWriter.name(JSONNAMES[10]).value(config.areDeweyIDsStored);
+      // Persistenter.
+      jsonWriter.name(JSONNAMES[11]).value(config.recordPersister.getClass().getName());
+      jsonWriter.endObject();
+    } catch (final IOException e) {
+      throw new SirixIOException(e);
+    }
+
+    // Database config.
+    DatabaseConfiguration.serialize(config.databaseConfig);
+  }
+
+  /**
+   * Deserializing a Resource configuration from a JSON-file from the persistent storage.
+   *
+   * @param file where the resource lies in.
+   * @return a complete {@link ResourceConfiguration} instance
+   * @throws SirixIOException if an I/O error occurs
+   */
+  public static ResourceConfiguration deserialize(final Path file) throws SirixIOException {
+    try {
+      final Path configFile = file.resolve(ResourcePaths.CONFIG_BINARY.getPath());
+      final FileReader fileReader = new FileReader(configFile.toFile());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *fileReader*. Currently, there are execution paths that do not contain closure statements. Either a) close *fileReader* in a try-finally block or b) close the resource by declaring *fileReader* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).